robottelo/cli/proxy.py

```diff
@@ -51,20 +54,27 @@ def default_url_on_new_port(oldport, newport):
 
     with ssh._get_connection() as connection:
         command = (
-            u'ssh -i {0} -t -t -o UserKnownHostsFile=/dev/null'
-            ' -o StrictHostKeyChecking=no -L {1}:{2}:{3} {4}@{5}'
+            u'ssh -i {0} -o StrictHostKeyChecking=no -R {1}:{2}:{3} {4}@{5}'
         ).format(
             '/tmp/dsa_{0}'.format(newport),
             newport, domain, oldport, user, domain)
         logger.debug('Creating tunnel {0}'.format(command))
-        # Run command and timeout in 30 seconds.
-        _, _, stderr = connection.exec_command(command, 30)
-
-        stderr = stderr.read()
-        if len(stderr) > 0:
-            logger.debug('Tunnel failed: {0}'.format(stderr))
-            # Something failed, so raise an exception.
-            raise SSHTunnelError(stderr)
+        transport = connection.get_transport()
+        channel = transport.open_session()
+        channel.exec_command(command)
+        # if exit_status appears until command_timeout, throw error
+        for _ in xrange(command_timeout):
+            if channel.exit_status_ready():
+                if channel.recv_exit_status() != 0:
+                    stderr = u''
+                    while channel.recv_stderr_ready():
+                        stderr += channel.recv_stderr(1)
+                    logger.debug('Tunnel failed: {0}'.format(stderr))
+                    # Something failed, so raise an exception.
+                    raise SSHTunnelError(stderr)
+            sleep(1)
+        # do the cleanup
+        ssh.command('rm -f /tmp/dsa_{0}'.format(newport))
```

Comment:
Isn't better to remove the key file after the yield? Also the comment above can be remove, `ssh.command('rm` already states a cleanup.